myia/infer/infnode.py

```diff
@@ -200,10 +200,15 @@ def infer_graph(graph, input_types):
     g = graph.specialize(input_types)
     eng.replacements[g] = {}
     res = infer(eng, g.return_)
+    # Allocate None entry before iterating on replacements.
+    # TODO Remove ?
+    _ = eng.replacements[None]
```

Comment:
Ok ! That's because I got a `RuntimeError: dictionary changed size during iteration` later in `eng.replacement.items()` call. But, this error should occur in CI tests too, so I will remove my changes here.

Reply:
Oh, I see. The usual fix to that is `for x in list(d.items())` instead of `for x in d.items()`, but in this case maybe it's better to initialize `eng.replacements[None]` to an empty dict in `__init__`, which is similar to what you're doing but less cryptic.

Reply:
Ok, done !